server/src/lib/model/clips.ts

```diff
@@ -145,10 +152,36 @@ export default class Clips {
     clip.sentenceText = text;
   }
 
+  private clipsWithFewVotes: DBClipWithVoters[] = [];
+  private isRefilling = false;
+  private async refillClipsWithFewVotes() {
+    if (this.isRefilling) return;
+    this.isRefilling = true;
+    this.clipsWithFewVotes = this.clipsWithFewVotes.concat(
+      Random.shuffle(
+        this.randomEngine,
+        await this.db.findClipsWithFewVotes(CLIP_CACHE_SIZE)
+      )
+    );
+    this.isRefilling = false;
+  }
+
   /**
    * Fetch a random clip but make sure it's not the user's.
    */
-  getEllibleClip(userid: string): Clip {
+  async getEllibleClip(client_id: string): Promise<DBClipWithVoters | Clip> {
+    if (this.isMigrated) {
+      if (!this.clipsWithFewVotes.length) await this.refillClipsWithFewVotes();
```

Comment:
nit: it's always better to be more semactic rather than rely on truthiness if possible, ie.:
```
if (this.clipsWithFewVotes.length < 1)